packages/openapi-code-generator/src/core/openapi-types-normalized.ts

```diff
@@ -68,11 +69,19 @@ export interface IRModelBoolean extends IRModelBase {
   enum?: string[] | undefined
 }
 
-export interface IRModelObject extends IRModelBase {
-  allOf: MaybeIRModel[]
```

Comment:
after this change, it's only the input normalization that has to deal with things being an object, composition or both at once.